src/cc/bcc_elf.c

```diff
@@ -911,6 +928,32 @@ int bcc_elf_get_buildid(const char *path, char *buildid)
   return 0;
 }
 
+int bcc_elf_symbol_str(const char *path, size_t section_idx,
+                       size_t str_table_idx, char *out, size_t len)
+{
+  Elf *e;
+  int fd, err = 0;
+  const char *name;
+
+  if (!out || !len)
+    return -1;
+
+  if (openelf(path, &e, &fd) < 0)
+    return -1;
+
+  if ((name = elf_strptr(e, section_idx, str_table_idx)) == NULL) {
+    err = -1;
+    goto exit;
+  }
+
+  strncpy(out, name, len);
```

Comment:
So that we can just refer to the string pointer here directly?

Reply:
Unless I misunderstand, not with the new changes.